### DISCUSSION TITLE
lvl=warn msg="Error compacting TSM files"

### DISCUSSION THREAD

**Discussion by JAK0721** [influxdata/influxdb#25016] [Q&A]
I have noticed that influxdb creating high CPU load
`PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND`
`6589 influxdb  20   0   14.0g 847056 621808 S  57.3  20.4  43:52.31 influxd`

I checked the journal with ` sudo journalctl --unit influxdb --since "30 min  ago"`

There are losts of entries including error messages like lvl=warn msg="Error compacting TSM files"

`May 20 12:23:17 raspberrypi influxd-systemd-start.sh[6589]: ts=2024-05-20T10:23:17.894833Z lvl=warn msg="Error compacting TSM files" log_id=0pHHzFyG000 service=storage-engine engine=tsm1 tsm1_level=2 tsm1_strategy=level op_name=tsm1_compact_group error="block read error on /var/lib/influxdb/engine/data/3f39039b536f555e/autogen/186/000000052-000000002.tsm: decode error: unable to decompress block type integer for key 'vehicleSoc,loadpoint=Garage#!~#value': invalid block type: exp 1, got 0, block read error on /var/lib/influxdb/engine/data/3f39039b536f555e/autogen/186/000000052-000000002.tsm: decode error: unable to decompress block type integer for key 'vehicleSoc,loadpoint=Garage#!~#value': invalid block type: exp 1, got 0, additional errors dropped: 0"
May 20 12:23:18 raspberrypi influxd-systemd-start.sh[6589]: ts=2024-05-20T10:23:18.895014Z lvl=info msg="TSM compaction (end)" log_id=0pHHzFyG000 service=storage-engine engine=tsm1 tsm1_level=2 tsm1_strategy=level op_name=tsm1_compact_group op_event=end op_elapsed=29667.806ms
May 20 12:23:19 raspberrypi influxd-systemd-start.sh[6589]: ts=2024-05-20T10:23:19.227570Z lvl=info msg="TSM compaction (start)" log_id=0pHHzFyG000 service=storage-engine engine=tsm1 tsm1_level=2 tsm1_strategy=level op_name=tsm1_compact_group op_event=start
May 20 12:23:19 raspberrypi influxd-systemd-start.sh[6589]: ts=2024-05-20T10:23:19.227587Z lvl=info msg="Beginning compaction" log_id=0pHHzFyG000 service=storage-engine engine=tsm1 tsm1_level=2 tsm1_strategy=level op_name=tsm1_compact_group tsm1_files_n=4
May 20 12:23:19 raspberrypi influxd-systemd-start.sh[6589]: ts=2024-05-20T10:23:19.227594Z lvl=info msg="Compacting file" log_id=0pHHzFyG000 service=storage-engine engine=tsm1 tsm1_level=2 tsm1_strategy=level op_name=tsm1_compact_group tsm1_index=0 tsm1_file=/var/lib/influxdb/engine/data/3f39039b536f555e/autogen/186/000000009-000000002.tsm
May 20 12:23:19 raspberrypi influxd-systemd-start.sh[6589]: ts=2024-05-20T10:23:19.227602Z lvl=info msg="Compacting file" log_id=0pHHzFyG000 service=storage-engine engine=tsm1 tsm1_level=2 tsm1_strategy=level op_name=tsm1_compact_group tsm1_index=1 tsm1_file=/var/lib/influxdb/engine/data/3f39039b536f555e/autogen/186/000000018-000000002.tsm
May 20 12:23:19 raspberrypi influxd-systemd-start.sh[6589]: ts=2024-05-20T10:23:19.227608Z lvl=info msg="Compacting file" log_id=0pHHzFyG000 service=storage-engine engine=tsm1 tsm1_level=2 tsm1_strategy=level op_name=tsm1_compact_group tsm1_index=2 tsm1_file=/var/lib/influxdb/engine/data/3f39039b536f555e/autogen/186/000000026-000000002.tsm
May 20 12:23:19 raspberrypi influxd-systemd-start.sh[6589]: ts=2024-05-20T10:23:19.227614Z lvl=info msg="Compacting file" log_id=0pHHzFyG000 service=storage-engine engine=tsm1 tsm1_level=2 tsm1_strategy=level op_name=tsm1_compact_group tsm1_index=3 tsm1_file=/var/lib/influxdb/engine/data/3f39039b536f555e/autogen/186/000000052-000000002.tsm
May 20 12:23:52 raspberrypi influxd-systemd-start.sh[6589]: ts=2024-05-20T10:23:52.151810Z lvl=warn msg="Error compacting TSM files" log_id=0pHHzFyG000 service=storage-engine engine=tsm1 tsm1_level=2 tsm1_strategy=level op_name=tsm1_compact_group error="block read error on /var/lib/influxdb/engine/data/3f39039b536f555e/autogen/186/000000052-000000002.tsm: decode error: unable to decompress block type integer for key 'vehicleSoc,loadpoint=Garage#!~#value': invalid block type: exp 1, got 0, block read error on /var/lib/influxdb/engine/data/3f39039b536f555e/autogen/186/000000052-000000002.tsm: decode error: unable to decompress block type integer for key 'vehicleSoc,loadpoint=Garage#!~#value': invalid block type: exp 1, got 0, additional errors dropped: 0"`

I am using a InfluxDB v2.7.6 on a RPI 5.
A search on the internet as well as asking ChatGBT was not really helpful. Any help would be highly appreciated!
Thanks in advance

**Comment by JAK0721** [comment]
solved the issue myself.
run  `sudo rm /var/lib/influxdb/engine/data/3f39039b536f555e/autogen/186/000000052-000000002.tsm`
 `sudo touch /forcefsck` & `sudo reboot`
Did this as often as errors occurs with other *.tsm files

**Reply by ghost** [reply]
did you loose some part of the data?
if, how did you manage to restore it? 
and how did you verify weather all the data was restored or not

**Comment by philjb** [comment]
Those kinda of tsm errors are usually tsm file corruption. Best to delete the tsm file as the author has done and recreate/restore from a replica or backup. Perhaps a single byte mistake could be corrected by guessing what the correct value should be, but i wouldn't only recommend that in case where this is the only copy of the data and the data is extremely important to recover. 

Closing per author's own resolution.
